doc/hooks/installShellFiles.section.md

```diff
@@ -17,11 +17,38 @@ The `installShellCompletion` function takes one or more paths to shell completio
     installShellCompletion --zsh --name _foobar share/completions.zsh
     # implicit behavior
     installShellCompletion share/completions/foobar.{bash,fish,zsh}
-    # using named fd
+  '';
+}
+```
+
+When generating completions requires the execution of programs,
+it's necessary to disable this process during cross-compilation.
+This is because the current build platform may not be able to run binaries of the host platform.
+
+```nix
+{
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    # Utilizing named file descriptor
     installShellCompletion --cmd foobar \
       --bash <($out/bin/foobar --bash-completion) \
       --fish <($out/bin/foobar --fish-completion) \
       --zsh <($out/bin/foobar --zsh-completion)
   '';
 }
 ```
+
```

Comment:
Some people have spoken out against this solution before https://github.com/NixOS/nixpkgs/pull/316749#issuecomment-2144047010

Reply:
I cannot quite follow the reasoning in this thread though. Qemu is not that bad to have in your build closure. It should be already compiled anyway. But I think it's harder to test, so I personally prefer if contributor at least remove this completion for cross compilation.

Reply:
I don't have a strong opinion either way, just trying to tie all the discussions in various places together (and hopefully the bikeshedding will converge on something).

Separate question: why is this in `preFixup`? Is that because it's before wrapping happens?

Reply:
I don't have a strong opinion here. I let you let write the documentation than.
Just ping me on the review.

Reply:
@Mic92 #325591 was already merged before I got to respond to your comment.